homeassistant/components/climate/generic_thermostat.py

```diff
@@ -185,14 +183,20 @@ def target_temperature(self):
     @property
     def operation_list(self):
         """List of available operation modes."""
-        return [STATE_AUTO, STATE_OFF]
+        return self._operation_list
 
     def set_operation_mode(self, operation_mode):
         """Set operation mode."""
-        if operation_mode == STATE_AUTO:
+        if operation_mode == STATE_HEAT:
+            self._current_operation = STATE_HEAT
```

Comment:
Setting operation mode doesn't seem to influence how the thermostat is operating, ie cooling or heating. So what's the point of enabling those options? Although `_enabled` can be replaced by `_current_operation != STATE_OFF`, so that selection will influence on/off.

Reply:
Operation mode allows two options (Based on the type of generic_thermostat): Either heating or off. I was trying to fix operation_mode discrepancy initially introduced by making it use the correct concepts. 

I plan to add the possibility to have both heater switch and cooling switch so you can have heating, cooling, off and also auto with the correct logic.

Reply:
Ah, ok, so currently it's still like an on/off setting basically.

Reply:
Yes, however now the bug there the "on" or "auto" was selected and not visible on the interface is gone. Now you clearly see "heat" selected on the interface. 

The only downside (which I have no solution yet) is that you lose the nice feedback on the interface when the heater is idle or heating, but the way it was represented now looked more like hack.